jerry-ext/handler/handler-assert.c

```diff
@@ -132,3 +160,213 @@ jerryx_handler_assert (const jerry_value_t func_obj_val, /**< function object */
 {
   return jerryx_handler_assert_fatal (func_obj_val, this_p, args_p, args_cnt);
 } /* jerryx_handler_assert */
+
+
+static jerry_value_t
+jerryx_handler_assert_equal_helper (const jerry_value_t func_obj_val, /**< function object */
+                                    const jerry_value_t this_p, /**< this arg */
+                                    const jerry_value_t args_p[], /**< function arguments */
+                                    const jerry_length_t args_cnt, /**< number of function arguments */
+                                    operations type)
+{
+  (void) func_obj_val; /* unused */
+  (void) this_p; /* unused */
+
+  if (args_cnt < 2)
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE, (const jerry_char_t *) "Expected at least 2 arguments.");
+  }
+
+  jerry_value_t actual = args_p[0];
+  jerry_value_t expected = args_p[1];
+
+  jerry_binary_operation_t op = JERRY_BIN_OP_EQUAL;
+  switch (type)
+  {
+    case JERRYX_OP_EQUAL:
+    case JERRYX_OP_NOT_EQUAL:
+    op = JERRY_BIN_OP_EQUAL;
+    break;
+
+    case JERRYX_OP_STRICT_EQUAL:
+    case JERRYX_OP_NOT_STRICT_EQUAL:
+    op = JERRY_BIN_OP_STRICT_EQUAL;
+    break;
+    // JERRY_UNREACHABLE ();
+  }
+
+  jerry_value_t result = jerry_binary_operation (op, actual, expected);
+  bool bool_result = jerry_get_boolean_value (result);
+
+  if (!jerry_value_is_boolean (result) || type > JERRYX_OP_STRICT_EQUAL ? bool_result : !bool_result)
```

Comment:
You should add a static assert due to this check `type > JERRYX_OP_STRICT_EQUAL`.